src/include/miopen/kernel.hpp

```diff
@@ -26,17 +26,16 @@
 #ifndef GUARD_MIOPEN_KERNEL_HPP
 #define GUARD_MIOPEN_KERNEL_HPP
 
-#include <string>
+#include <string_view>
 #include <vector>
 
 #include <miopen/config.h>
+#include <miopen/filesystem.hpp>
 
 namespace miopen {
-std::string GetKernelSrc(std::string name);
-std::string GetKernelInc(std::string key);
-const std::string* GetKernelIncPtr(std::string key);
-std::vector<std::string> GetKernelIncList();
-std::vector<std::string> GetHipKernelIncList();
+std::string_view GetKernelSrc(const fs::path& name);
+std::string_view GetKernelInc(const fs::path& name);
+const std::vector<std::reference_wrapper<const fs::path>>& GetKernelIncList();
```

Comment:
If we are going to keep `fs::path` as a "keys" in the "database of kernel sources" then I so not see any good in resolving comments from my previous review. Maybe it would be even better to keep using fs:path in comgr etc.

I know that @DrizztDoUrden is thinking that `fs:path` is good, and I am not going to argue endlessly. And finally you have my "minority report".